node/tests/TestUtilities.ts

```diff
@@ -51,6 +51,35 @@ export function getFirstResult(
     return Object.values(res).at(0);
 }
 
+export function compareMaps(
+    map: Record<string, unknown>,
+    map2: Record<string, unknown>,
+) {
+    // Check if both maps have the same keys
+    expect(Object.keys(map)).toEqual(Object.keys(map2));
+
+    // Check if each key in the first map has the same value in the second map
+    for (const key in map) {
+        if (
+            map[key] !== null &&
+            typeof map[key] === "object" &&
+            map2[key] !== null &&
+            typeof map2[key] === "object" &&
+            !Array.isArray(map[key]) &&
+            !Array.isArray(map2[key])
```

Comment:
this is really not readable.
I think `map[key] !== null` can be removed if we check for `typeof map2[key] === "object"`.
Try to check if there's a better way of checking if it's a map, and add a comment explaining what you do there